File: MobileApp/python/events_events.py
from req_lib import ReqLib
import datetime

'''
This endpoint on the MobileApp API returns 
an iCal stream of dining venue open hours.
The iCal stream really is just a plain 
text stream.

Parameters: from, to

The from and to parameters are dates formatted
in the following way:

YYYYMMDD

'''

if __name__ == "__main__":
    req_lib = ReqLib()
    # Keep in mind that because "from" is a 
    # keyword in the python language, we 
    # can't put "from" as a parameter in 
    # the following function

    # Get today's date
    today = datetime.datetime.today()
    year = str(today.year)

    # Pad the number with zeros so that 
    # there are always exactly two digits
    month = str(today.month).zfill(2)
    day = str(today.day).zfill(2)

    # Get the date 2 days from now
    two_days_later = today + datetime.timedelta(days=2)
    new_year = str(two_days_later.year)

    # Pad the number with zeros so that 
    # there are always exactly two digits
    new_month = str(two_days_later.month).zfill(2)
    new_day = str(two_days_later.day).zfill(2)

    req = req_lib.getXMLorTXT(
        req_lib.configs.EVENTS_EVENTS,
        kwargs={
            "from": year + month + day,
            "to": new_year + new_month + new_day,
        }
    )
    print(req)

File: MobileApp/java/EventsEvents.java
import java.util.HashMap;
import java.time.LocalDate;

/*
This endpoint on the MobileApp API returns 
an iCal stream of dining venue open hours.
The iCal stream really is just a plain 
text stream.

Parameters: from, to

The from and to parameters are dates formatted
in the following way:

YYYYMMDD

*/

public class EventsEvents {
    public static void main(String[] args) {
        ReqLib reqLib = new ReqLib();
        String endpoint = reqLib.configs.events_events;
        HashMap<String, Object> params = new HashMap<String, Object>();

        // Extract the date
        String[] date = LocalDate.now().toString().split("-");
        String year = date[0];
        String month = date[1];
        String day = date[2];

        // Calculate the date two days from now
        String[] two_days_later = LocalDate.now().plusDays(2).toString().split("-");
        String new_year = two_days_later[0];
        String new_month = two_days_later[1];
        String new_day = two_days_later[2];

        params.put("from", year + month + day);
        params.put("to", new_year + new_month + new_day);
        
        try {
            String events = reqLib.getRequest(endpoint, params);
            System.out.println(events);
        } catch (Exception e) {
            throw new RuntimeException(e);
        }
    }
}